python/pfs/drp/stella/fitPfsFluxReference.py

```diff
@@ -721,7 +710,37 @@ def fitModelsToSpectra(
                 whitenedModel=whitenedModels,
             )
 
-        return chisqLists
+        if self.debugInfo.doWriteChisq:
+            debugging.writeExtraData(
+                f"fitPfsFluxReference-output/chisq-{obsSpectra.filename}.pickle",
+                fiberId=pfsConfig.fiberId,
+                chisq=chisqLists,
+            )
+
+        # Posterior PDF
+        pdfs: List[Union[NDArray[np.float64], None]] = []
+        for prior, chisq in zip(priorPdfs, chisqLists):
```

Comment:
The problem is that the double loop is not `for each observed spectrum; for each model; do...` but `for each model; for each observed spectrum; do...`. The double loop must be in this order because it takes more time to prepare a model than to prepare an observed spectrum for comparison. Once the brute-force method is gone, I think it better for there to be a Task that fit a single fiber, for it would pave the way to parallelization.